src/file.py

```diff
@@ -0,0 +1,338 @@
+"""
+file.py
+
+File class to interact with the CS3 API.
+"""
+
+import time
+import logging
+import http
+import requests
+import cs3.storage.provider.v1beta1.resources_pb2 as cs3spr
+import cs3.storage.provider.v1beta1.provider_api_pb2 as cs3sp
+from cs3.gateway.v1beta1.gateway_api_pb2_grpc import GatewayAPIStub
+import cs3.rpc.v1beta1.code_pb2 as cs3code
+import cs3.types.v1beta1.types_pb2 as types
+
+from config import Config
+from typing import Iterator
+from exceptions.exceptions import AuthenticationException, FileLockedException, NotFoundException
+from cs3resource import Resource
+from auth import Auth
+
+
+class File:
```

Comment:
Maybe we should add in the docstrings that the result of some requests is None if all is well, or throws an exception if failed.